core/input/util.js

```diff
@@ -1,679 +1,170 @@
-
-        if (keycode >= 0x70 && keycode <= 0x87) {
-            return KeyTable.XK_F1 + keycode - 0x70; // F1-F24
-        }
-        switch (keycode) {
-
-            case 8 : return KeyTable.XK_BackSpace;
-            case 13 : return KeyTable.XK_Return;
-
-            case 9 : return KeyTable.XK_Tab;
-
-            case 27 : return KeyTable.XK_Escape;
-            case 46 : return KeyTable.XK_Delete;
-
-            case 36 : return KeyTable.XK_Home;
-            case 35 : return KeyTable.XK_End;
-            case 33 : return KeyTable.XK_Page_Up;
-            case 34 : return KeyTable.XK_Page_Down;
-            case 45 : return KeyTable.XK_Insert;
-
-            case 37 : return KeyTable.XK_Left;
-            case 38 : return KeyTable.XK_Up;
-            case 39 : return KeyTable.XK_Right;
-            case 40 : return KeyTable.XK_Down;
-
-            case 16 : return KeyTable.XK_Shift_L;
-            case 17 : return KeyTable.XK_Control_L;
-            case 18 : return KeyTable.XK_Alt_L; // also: Option-key on Mac
-
-            case 224 : return KeyTable.XK_Meta_L;
-            case 225 : return KeyTable.XK_ISO_Level3_Shift; // AltGr
-            case 91 : return KeyTable.XK_Super_L; // also: Windows-key
-            case 92 : return KeyTable.XK_Super_R; // also: Windows-key
-            case 93 : return KeyTable.XK_Menu; // also: Windows-Menu, Command on Mac
-            default: return null;
-        }
-    }
-
-    KeyboardUtil.hasShortcutModifier = hasShortcutModifier;
-    KeyboardUtil.hasCharModifier = hasCharModifier;
-    KeyboardUtil.ModifierSync = ModifierSync;
-    KeyboardUtil.getKey = getKey;
-    KeyboardUtil.getKeysym = getKeysym;
-    KeyboardUtil.keysymFromKeyCode = keysymFromKeyCode;
-    KeyboardUtil.nonCharacterKey = nonCharacterKey;
-    KeyboardUtil.substituteCodepoint = substituteCodepoint;
-})();
-
-KeyboardUtil.QEMUKeyEventDecoder = function(modifierState, next) {
-    "use strict";
-
-    function sendAll(evts) {
-        for (var i = 0; i < evts.length; ++i) {
-            next(evts[i]);
-        }
-    }
 
-    var numPadCodes = ["Numpad0", "Numpad1", "Numpad2",
-        "Numpad3", "Numpad4", "Numpad5", "Numpad6",
-        "Numpad7", "Numpad8", "Numpad9", "NumpadDecimal"];
-
-    var numLockOnKeySyms = {
-        "Numpad0": 0xffb0, "Numpad1": 0xffb1, "Numpad2": 0xffb2,
-        "Numpad3": 0xffb3, "Numpad4": 0xffb4, "Numpad5": 0xffb5,
-        "Numpad6": 0xffb6, "Numpad7": 0xffb7, "Numpad8": 0xffb8,
-        "Numpad9": 0xffb9, "NumpadDecimal": 0xffac
-    };
-
-    var numLockOnKeyCodes = [96, 97, 98, 99, 100, 101, 102,
-        103, 104, 105, 108, 110];
-
-    function isNumPadMultiKey(evt) {
-        return (numPadCodes.indexOf(evt.code) !== -1);
-    }
-
-    function getNumPadKeySym(evt) {
-        if (numLockOnKeyCodes.indexOf(evt.keyCode) !== -1) {
-            return numLockOnKeySyms[evt.code];
+        // We start with layout independent keys
+        var code = getKeycode(evt);
```

Comment:
We should, but unfortunately support for `key` isn't where we need it to be yet. As I mentioned in the other comment, this step is only for keys that are the same on every layout. So it will not do anything for e.g. `A` as that might be mapped to something else.